ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/LedgerDB/V1/BackingStore/Registry.hs

```diff
@@ -0,0 +1,62 @@
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GeneralisedNewtypeDeriving #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE Rank2Types #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | A utility for storing and retrieving resources in a registry using handles
+-- to identify resources in the registry.
+module Test.Ouroboros.Storage.LedgerDB.V1.BackingStore.Registry (
+    Handle
+  , HandleRegistry
+  , initHandleRegistry
+  , readHandle
+  , registerHandle
+  ) where
+
+import           Control.Monad.Class.MonadSTM.Internal as STM
+                     (MonadSTM (TVar, atomically, newTVarIO, readTVar, writeTVar))
+import           Data.Map.Strict (Map)
+import qualified Data.Map.Strict as Map
+import           Ouroboros.Consensus.Util.IOLike (IOLike)
+
+newtype Handle = Handle Word
+  deriving stock (Show, Eq, Ord)
+  deriving newtype Num
```

Comment:
This `Num` instance lets folks create eg `(5 :: Handle)`. You probably don't want that, since `Handle` is exported opaquely.